zeebe/exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/ElasticsearchExporterIT.java

```diff
@@ -325,11 +355,94 @@ final var record = factory.generateRecord(ValueType.JOB);
       }
     }
 
+    @Test
+    void shouldExportRecordToIndexSpecifiedByRecordBrokerVersion() {
+      configureExporter(false);
+      final var oldRecord =
+          factory.generateRecord(ValueType.JOB, r -> r.withBrokerVersion("8.6.0"));
+
+      try (final var mockVersion =
+          Mockito.mockStatic(VersionUtil.class, Mockito.CALLS_REAL_METHODS)) {
+        mockVersion.when(VersionUtil::getVersionLowerCase).thenReturn("8.7.0");
+
+        await("New record is exported, and existing indices are updated")
+            .atMost(Duration.ofSeconds(30))
+            .until(() -> export(oldRecord));
+      }
+
+      final var document = testClient.getExportedDocumentFor(oldRecord);
+      assertThat(document.index().contains(oldRecord.getBrokerVersion())).isTrue();
+    }
+
     private void configureExporter(final boolean retentionEnabled) {
       config.retention.setEnabled(retentionEnabled);
       exporter.configure(exporterTestContext);
     }
 
+    @Test
+    void shouldExportToCorrectIndexWithElasticsearchNotReachable() throws IOException {
+
+      // given
+      final var currentPort = CONTAINER.getFirstMappedPort();
+      CONTAINER.stop();
+      Awaitility.await().until(() -> !CONTAINER.isRunning());
+
+      final var record = factory.generateRecord(r -> r.withBrokerVersion("8.6.0"));
+
+      try (final var mockVersion =
+          Mockito.mockStatic(VersionUtil.class, Mockito.CALLS_REAL_METHODS)) {
+        mockVersion.when(VersionUtil::getVersionLowerCase).thenReturn("8.6.0");
+        configureExporter(false);
+
+        assertThatThrownBy(() -> export(record));
+      }
+
+      CONTAINER
+          .withEnv("discovery.type", "single-node")
+          .setPortBindings(List.of(currentPort + ":9200"));
+      CONTAINER.start();
+      Awaitility.await().until(CONTAINER::isRunning);
+
+      // when
+      final var record2 = factory.generateRecord(r -> r.withBrokerVersion("8.7.0"));
+      try (final var mockVersion =
+          Mockito.mockStatic(VersionUtil.class, Mockito.CALLS_REAL_METHODS)) {
+        mockVersion.when(VersionUtil::getVersionLowerCase).thenReturn("8.7.0");
+        configureExporter(false);
```

Comment:
❓ What does this do?

Reply:
oops yea it doesn't need to be configured twice as the port doesn't change I'll move it out.